libraries/standard/http/src/http_client.c

```diff
@@ -1608,6 +1604,10 @@ static HTTPStatus_t sendHttpData( const TransportInterface_t * pTransport,
         }
         else
         {
+            /* It is a bug in the application's transport send implementation if
+             * more bytes than expected are sent. */
+            assert( ( size_t ) transportStatus <= bytesRemaining );
```

Comment:
This is moved to fix a possible overflow in unsigned to signed conversion.

Reply:
Please mention this in comment so that someone does not try to revert it later.

Reply:
Sure